### PR TITLE
Add limit to call depth and support assert_exhaustion

### PR DESCRIPTION
**Pull request by gumb0**
https://github.com/WebAssembly/spec/issues/1060
https://github.com/WebAssembly/spec/blob/6d6294fb456ea941b2c1b0346919392dbb023be1/test/core/call.wast#L96-L103

### CODE REVIEW

File: lib/fizzy/limits.hpp [hunk 1]
```diff
@@ -5,4 +5,6 @@ namespace fizzy
 constexpr unsigned PageSize = 65536;
 // Set hard limit of 256MB of memory.
 constexpr unsigned MemoryPagesLimit = (256 * 1024 * 1024ULL) / PageSize;
+
+constexpr int CallStackLimit = 2048;
```
Comment (axic) [inline lib/fizzy/limits.hpp:9]: Can you add a comment where this came from? And a `TODO: review this` note?

**Comment by axic** [comment]
Can you perhaps squash the first two commits?

And also test if #258 still pass with this limit?

File: test/unittests/execute_call_test.cpp [hunk 2]
```diff
@@ -341,3 +341,33 @@ TEST(execute_call, imported_function_from_another_module)
...
+      )
+    */
+    const auto bin = from_hex(
+        "0061736d010000000105016000017f03020100040501700101010907010041000b01000a090107004100110000"
+        "0b");
+
+    const Module module = parse(bin);
+
+    EXPECT_TRUE(execute(module, 0, {}).trapped);
+}
```
Comment (chfast) [inline test/unittests/execute_call_test.cpp:373]: Add test for `execute(..., 2048)`.

File: lib/fizzy/execute.cpp [hunk 3]
```diff
@@ -322,15 +322,18 @@ const FuncType& function_type(const Instance& instance, FuncIdx idx)
     return instance.module.typesec[type_idx];
 }
 
-bool invoke_function(
-    const FuncType& func_type, uint32_t func_idx, Instance& instance, Stack<uint64_t>& stack)
+bool invoke_function(const FuncType& func_type, uint32_t func_idx, Instance& instance,
+    Stack<uint64_t>& stack, int depth)
 {
+    if (depth == CallStackLimit)
```
Comment (axic) [inline lib/fizzy/execute.cpp:328]: Add `assert(depth >= 0)`.

File: test/unittests/execute_call_test.cpp (hunk 2 above)
Comment (gumb0) [reply test/unittests/execute_call_test.cpp:373]: added below

File: lib/fizzy/execute.cpp (hunk 3 above)
Comment (gumb0) [reply lib/fizzy/execute.cpp:328]: I added it to `execute`

**Comment by gumb0** [comment]
@axic fibonacci and factorial still work (see CI)

**Pull request merged**